### PR TITLE
feat: create organisation by users

### PR DESCRIPTION
**Pull request by victoradepoju**
## How should this be manually tested?
- Try accessing the `/api/v1/organisations` route as an authorized user (Bearer Token).
- Submit a valid request to create an organisation.
- The expected result is to get a `201 Created` response with the organisation details.
- Error responses to check:
  - `403 Unauthorized` if the user is not authenticated.
  - `422 Unprocessable Entity` if there are validation errors.
  -  `400 Bad Request` if the organisation name already exists.

## Checklist of what you did
- [x] Implemented authentication middleware to ensure the user is authenticated using JWT.
- [x] Created the `/api/v1/organisations` endpoint to handle organisation creation.
- [x] Added input validation for all supplied fields in the request body.
- [x] Implemented error handling for various scenarios including name conflicts, validation errors, unauthorized access, and internal server errors.
- [x] Ensured the response structure is consistent for successful and error scenarios.
- [x] Added unit tests and e2e tests for all the features as described more in the reference issue below.
## Reference Issue
[https://github.com/hngprojects/hng_boilerplate_java_web/issues/21]()

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,7 +1,10 @@
 spring.application.name=hng-java-boilerplate
 
 # datasource configuration
-spring.datasource.url=jdbc:${connection_string}
+# spring.datasource.url=jdbc:postgresql://localhost:5432/postgres
```
Comment (Fifetoyi) [inline src/main/resources/application.properties:4]: remove comments. local test changes should not be pushed

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -30,4 +33,30 @@ spring.datasource.hikari.leak-detection-threshold=15000
 spring.mvc.throw-exception-if-no-handler-found=true
 spring.web.resources.add-mappings=false
 
-sha512.string=5627dcc7bb45a7cf9ddd211168fb727a2005338c5fe1e29faa71fc542bcd59e1bf0c604d65578ba9001984bfa0f25e29a509f130db33c51916bf0ebfbb78b645
+#Logging
+spring.main.banner-mode=off
+logging.pattern.console = ${LOGGING_PATTERN: %green(%d{HH:mm:ss.SSS}) %blue(%-5level) %red([%thread]) %yellow(%logger{15})- %msg%n}
+
+#Swagger doc
+springdoc.api-docs.path=/docs/yml
```
Comment (Fifetoyi) [inline src/main/resources/application.properties:41]: @lurldgbodex please confirm that these changes are not already implemented

**Review by Fifetoyi — CHANGES_REQUESTED** [review]
changes required

File: src/main/java/hng_java_boilerplate/organisation/service/OrganisationService.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class OrganisationService {
+    private final OrganisationRepository organisationRepository;
+    private final UserRepository userRepository;
+
+    public CreateOrganisationResponseDto create(
+            CreateOrganisationRequestDto orgRequest,
```
Comment (lurldgbodex) [inline src/main/java/hng_java_boilerplate/organisation/service/OrganisationService.java:25]: why not do this using transaction to ensure atomicity.

File: src/main/resources/application.properties (hunk 2 above)
Comment (lurldgbodex) [reply src/main/resources/application.properties:41]: @Am0du  has majorly been doing the configurations with the devops guy. I've tagged him.

File: src/main/resources/application.properties (hunk 1 above)
Comment (victoradepoju) [reply src/main/resources/application.properties]: changed application.properties back to default

**Comment by victoradepoju** [comment]
@lurldgbodex @Fifetoyi kindly confirm the new changes.

**Comment by victoradepoju** [comment]
Just increased migration file version.

File: src/main/java/hng_java_boilerplate/organisation/service/OrganisationService.java (hunk 3 above)
Comment (victoradepoju) [reply src/main/java/hng_java_boilerplate/organisation/service/OrganisationService.java:27]: made create organisation transactional.

File: src/test/java/hng_java_boilerplate/organisation/service/OrganisationServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                "info@vpay.com",
+                "Finance",
+                "Private",
+                "USA",
+                "123 Payment St.",
+                "CA"
+        );
+        user = new User();
+        user.setId("user-123");
+//        when(activeUser.getPrincipal()).thenReturn(user);
```
Comment (Fifetoyi) [inline src/test/java/hng_java_boilerplate/organisation/service/OrganisationServiceTest.java:65]: remove if not needed

**Review by Fifetoyi — DISMISSED** [review]
lgtm

**Comment by lurldgbodex** [comment]
LGTM

**Comment by victoradepoju** [comment]
@lurldgbodex @Fifetoyi @Am0du all ci test are passing now, and got two lgtm.
Time to merge?

**Review by Am0du — APPROVED** [review]
LGTM

**Pull request merged**